src/action_upgrade.rs

```diff
@@ -93,12 +93,13 @@ pub fn upgrade(rua_env: &RuaEnv, devel: bool, printonly: bool) {
 		} else {
 			print_outdated(&outdated, &unexistent);
 			eprintln!();
+			let paths = rua_paths::RuaPaths::new();
```

Comment:
Hm, this is minor, but are we sure we want to check for
other instances conditionally in `rua upgrade`?
This might be surprising to users.

Reply:
Well the other scenario is the `rua upgrade --printonly`. Is it bad to run something in parallel with it?

This subcommand will fetch something from AUR and check the local database, but that's it. It doesn't change state, and it exists immediately (well, once HTTP things are done). Thoughts?

Reply:
Yeah, I guess you're right. `--printonly` should be fine without the checks.

I think what irks me is that both `action_install()` and `action_builddir()`
take `&RuaPaths` as an argument, but `action_upgrade()` does not
and instead constructs one internally. This hides the fact that the root/instance
checks are performed.

How about something like this?:
```rust
//main.rs:
if (printonly) {
  action::upgrade_printonly(devel);
} else {
  let rua_paths = RuaPaths::initialize_paths();
  action::upgrade(&rua_paths, devel);
}

//action_upgrade.rs:
fn some_common_upgrade_code();

fn upgrade() {
  some_common_upgrade_code();
  //...upgrade-specific code...
}

fn upgrade_printonly() {
  some_common_upgrade_code();
  //..printonly specific code...
}
```

Reply:
*sigh* you're right of course. Even though it's a bit messy to actually implement.

I've played around with creating common method and things on top of it, but it was a bit unpleasant. I've then tried to employ an enum to handle the action:
```
enum UpgradeType {
	Printonly,
	Real { paths: RuaPaths },
}
```
which allows things to be a bit more concise. WDYT?

Reply:
pushed the changes --^

Reply:
Hm, I think we can do better.

I took a quick stab at this at [draft/rua-lazy-paths](https://github.com/VukoDrakkeinen/rua/commit/17c862fc710606edf87939e2de0cab8e95f70340), and you're right that it's somewhat
messy. There is some minor duplication here and there, and the return types are fugly.

Still, I think that it would be the right way going forward. The enum way feels like a workaround.

Reply:
(Man, I hope I'm not being overly harsh here)

Reply:
I'd say the approaches are very similar already, but you were opinionated, and there's no problem for me with going for the latter approach, so merged that one in. Will make a release now so that `rua upgrade --printonly` works from root.

Reply:
Cool, that also works for me :+1: